src/main/java/org/wildfly/security/auth/realm/ldap/LdapSecurityRealm.java

```diff
@@ -147,17 +153,29 @@ public void registerIdentityChangeListener(Consumer<Principal> listener) {
         DirContext dirContext = null;
         try {
             dirContext = obtainContext();
-            EventContext eventContext = (EventContext) dirContext.lookup("");
-            eventContext.addNamingListener("", EventContext.SUBTREE_SCOPE, new ServerNotificationListener(listener));
+            innerRegisterIdentityChangeListener(dirContext,listener);
         } catch (Exception cause) {
-            throw log.ldapRealmFailedRegisterListener(cause);
+            //either connection died or realm not available during boot
+            synchronized (this.listenersPendingRegistration) {
+                //we need to wait, lets store
+                listenersPendingRegistration.add(listener);
+            }
+            log.ldapRealmDeferRegistration(this.name);
+            if(log.isDebugEnabled()){
+                log.debug("Listener registration failure: ", cause);
+            }
         } finally {
             if (dirContext != null) {
                 closeContext(dirContext);
             }
         }
     }
 
+    private void innerRegisterIdentityChangeListener(final DirContext dirContext, final Consumer<Principal> listener) throws NamingException {
```

Comment:
heh, what I meant is that this PR/JIRA does not warrant change in other resources. Change that this PR has is there in order to be able to identify realm which is not "fully up".
WRT to method name, there is "registerIdentityChangeListener", second one is private, if you want I can prefix it with _ if this is more suitable style.